### DISCUSSION TITLE
Cannot select effect color on KDE

### DISCUSSION THREAD

**Discussion by vitotankian** [Schneegans/Burn-My-Windows#177] [Q&A] [answered]
Hey just wanted to come say Thanks for the work and the port to Kwin. I have been seriously thinking on moving to Gnome solely because of Burn My Windows. Like, seriously.

Something weird happens in Kwin thou. Seems that when you try changing any effect color, nothing happens. No color option is shown actually, and the color is set to Black. Then if you try to go Default, nothing. The effect gets stuck with black as the color.

Maybe its just me. Anyway. Thanks!

**Comment by Schneegans** [comment]
Thanks for the kind words! Does this happen for all effects or just for specific ones? And have you tried the latest version from the `main` branch (with the upcoming _Incinerate_ effect)?

**Comment by vitotankian** [comment]
Yeah, actually incinerate is my main effect, love it!
I don't want to test this in all effects since they don't look as good as they do with default color. I did tried with three of them, with the same result.

**Comment by vitotankian** [comment]
I did a quick testing with all effects (except incinerate) and:

- Color cannot be changed.
- After I try to change the color, the effect behaves as if Black was the defined color
- Setting Default on each effect reastores every setting but the color

**Comment by Schneegans** [comment]
For me (KDE Neon) it simply works. I do not have any knowledge about the internals - I don't even know where KWin actually stores these settings. I'll try to learn as much as I can but we may have to wait until someone shows up who has the same issue. Then we could try to compare your systems...

**Reply by marozsas** [reply]
Yeah, the same to me. No color selection or whatever it is supposed to show.

My specs:
Operating System: openSUSE Tumbleweed 20220624
KDE Plasma Version: 5.25.1
KDE Frameworks Version: 5.95.0
Qt Version: 5.15.2
Kernel Version: 5.18.4-1-default (64-bit)
Graphics Platform: X11
Processors: 8 × Intel® Core™ i7-3770 CPU @ 3.40GHz
Memory: 15.6 GiB of RAM
Graphics Processor: NVIDIA GeForce GTX 970/PCIe/SSE2
Manufacturer: Gigabyte Technology Co., Ltd.

**Reply by Schneegans** [reply]
@marozsas Can you maybe try to run `journalctl -f -o cat` in a terminal and watch its output while you try to change the color? Maybe an error message gets logged...

**Comment by vitotankian** [comment]
I'm testing Ubuntu now, and I think I will stay in gnome hehe In Gnome all effect work beutifully

**Comment by Sodivad** [comment]
KWin uses QUiLoader to load the .ui files. 
For non-stock widgets like KColorButton it needs plugin files to load the widgets.
On Debian derivatives they are sadly included in the `-dev` packages. So you need to have ` libkf5widgetsaddons-dev` installed so the color button works.

**Reply by marozsas** [reply]
On opensuse tumbleweed there is no libKF5WidgetsAddons5-dev  (or *-devel as usual on TW)
I already have both libKF5WidgetsAddons5-5.95.0-1.1.x86_64 and libKF5WidgetsAddons5-lang-5.95.0-1.1.noarch installed at the time of this issue. So, may be in another package ?

**Reply by Sodivad** [reply]
I think on opensuse it's called `kwidgetsaddons-devel`.

The file you want is called `kwidgetsaddons5widgets.so ` and should in the qt plugins directory, i.e. some path that ends in 
`plugins/designer/kwidgetsaddons5widgets.so `

**Comment by marozsas** [comment, marked as answer]
Thank you !
I confirm that after installing kwidgetsaddons-devel-5.95.0-1.1.x86_64 solved the issue !
Now, clicking on the color button, the appropriate window is showed to select a color from a grid.
